### PR TITLE
Create and verify hbbft seals.

### PR DESCRIPTION
**Pull request by afck**

### CODE REVIEW

File: ethcore/hbbft_engine/src/hbbft_engine.rs [hunk 1]
```diff
@@ -64,7 +76,7 @@ const DEFAULT_DURATION: Duration = Duration::from_secs(1);
 impl TransitionHandler {
 	/// Returns the approximate time duration between the latest block and the minimum block time
 	/// or a keep-alive time duration of 1s.
-	fn duration_remaining_since_last_block(&self, client: Arc<EngineClient>) -> Duration {
+	fn duration_remaining_since_last_block(&self, client: Arc<dyn EngineClient>) -> Duration {
```
Comment (vgtom) [inline ethcore/hbbft_engine/src/hbbft_engine.rs]: Yes, the addition of dyn keyword is a good idea...makes things more readable

File: ethcore/hbbft_engine/src/hbbft_engine.rs [hunk 2]
```diff
@@ -435,24 +551,42 @@ impl Engine<EthereumMachine> for HoneyBadgerBFT {
 	}
 
 	fn sealing_state(&self) -> SealingState {
-		SealingState::Ready
+		// Purge obsolete sealing processes.
+		let next_block = match self
+			.client_arc()
```
Comment (vgtom) [inline ethcore/hbbft_engine/src/hbbft_engine.rs]: I kind of feel, the first method should be part of the object...like self.client_arc() and then others can be on next line

File: ethcore/hbbft_engine/src/hbbft_engine.rs (hunk 1 above)
Comment (afck) [reply ethcore/hbbft_engine/src/hbbft_engine.rs]: It's also a compiler warning now. :+1:

File: ethcore/hbbft_engine/src/hbbft_engine.rs (hunk 2 above)
Comment (afck) [reply ethcore/hbbft_engine/src/hbbft_engine.rs]: You mean something like this?
```rust
let client = match self.client_arc() {
    None => return SealingState::NotReady,
    Some(client) => client,
};
let next_block = match client.block_number(BlockId::Latest) {
    None => return SealingState::NotReady,
    Some(block_num) => block_num + 1,
};
```

Comment (afck) [reply ethcore/hbbft_engine/src/hbbft_engine.rs]: Done.

**Comment by afck** [comment]
I addressed the review comment and fixed the seals: I hadn't realized that the items in `Seal::Regular` need to be RLP-encoded themselves.

**Comment by dforsten** [comment]
Awesome, will test it with the latest fix!

**Pull request merged**